### ISSUE TITLE
t/sql.t fails (with older SQL::Abstract?)

### ISSUE THREAD

**Issue by eserte** [jhthorsen/mojo-mysql#49]
On some of my smokers:
```
#   Failed test 'right query'
#   at t/sql.t line 27.
#     Structures begin differing at:
#          $got->[0] = 'INSERT INTO `foo` ( `bar`) VALUES ( ? ) ON DUPLICATE KEY UPDATE _UPDATE_SET_VALUES SET `c` = ?'
#     $expected->[0] = 'INSERT INTO `foo` ( `bar`) VALUES ( ? ) ON DUPLICATE KEY UPDATE `c` = ?'

#   Failed test 'right query'
#   at t/sql.t line 41.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo` WHERE ( `bar` = ? )0'
#     $expected->[0] = 'SELECT * FROM `foo` WHERE ( `bar` = ? ) ORDER BY `yada` DESC'

#   Failed test 'right query'
#   at t/sql.t line 45.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`2'
#     $expected->[0] = 'SELECT * FROM `foo` LIMIT ? OFFSET ?'

#   Failed test 'right query'
#   at t/sql.t line 49.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`0'
#     $expected->[0] = 'SELECT * FROM `foo` GROUP BY bar, baz'

#   Failed test 'right query'
#   at t/sql.t line 52.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`0'
#     $expected->[0] = 'SELECT * FROM `foo` GROUP BY `bar`, `baz`'

#   Failed test 'right query'
#   at t/sql.t line 56.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`1'
#     $expected->[0] = 'SELECT * FROM `foo` GROUP BY `bar` HAVING `baz` = ?'

#   Failed test 'right query'
#   at t/sql.t line 60.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo` WHERE ( `bar` > ? )1'
#     $expected->[0] = 'SELECT * FROM `foo` WHERE ( `bar` > ? ) GROUP BY `bar` HAVING `baz` < ?'

#   Failed test 'right query'
#   at t/sql.t line 68.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`0'
#     $expected->[0] = 'SELECT * FROM `foo` FOR UPDATE'

#   Failed test 'right query'
#   at t/sql.t line 72.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`0'
#     $expected->[0] = 'SELECT * FROM `foo` LOCK IN SHARE MODE'

#   Failed test 'right query'
#   at t/sql.t line 75.
#     Structures begin differing at:
#          $got->[0] = 'SELECT * FROM `foo`0'
#     $expected->[0] = 'SELECT * FROM `foo` FOR SHARE'
# Looks like you failed 10 tests of 22.
t/sql.t .................... 
Dubious, test returned 10 (wstat 2560, 0xa00)
Failed 10/22 subtests 
```
This seems to happen if SQL::Abstract is older (i.e. not 1.85 or 1.86).

**Comment by jhthorsen** [comment]
@Tekki: Can you tell me, or make a PR with which version of SQL::Abstract is required?

https://github.com/jhthorsen/mojo-mysql/blob/master/cpanfile#L3

**Comment by Tekki** [comment]
Don't know for sure. The order_by stuff is stolen from mojo-pg and there we have 1.86 in the makefile (https://github.com/mojolicious/mojo-pg/blob/master/Makefile.PL#L33). Sri has made some pull requests at SQL::Abstract, so it's quite sure that we need 1.86 too.

**Comment by Tekki** [comment]
What about Mojolicious 7.55 in the cpanfile? Don't we need a more recent version?
From 8.02 I get warnings from Mojo::Promise in database.t that disappear in 8.09.
> t/database.t ................. 1/?      (in cleanup)    (in cleanup)  at .../site_perl/5.28.0/Mojo/Promise.pm line 91 during global destruction.

**Comment by jhthorsen** [comment]
I hope this gets fixed in 1.12.
